src/basins/basins_utilities.jl

```diff
@@ -155,3 +220,33 @@ function convergence_and_basins_fractions(mapper::AttractorMapper, ics::ValidICS
     ffs = Dict(k => v/N for (k, v) in fs)
     return ffs, labels, iterations
 end
+
+"""
+    map_to_basin(BoA::BasinOfAttraction, point; kwargs...) → id
+
+Given a `point`, `map_to_basin` interpolates to which basin it should belong.
+
+For `ArrayBasinsOfAttraction` this finds the label of the closest grid cell. 
+
+For `SampledBasinsOfAttraction` this finds the label corresponding to the 
+nearest neighbor using `Neighborhood.jl`, in which case the keyword arguments are:
+* `tree`: search tree constructor (e.g. `KDTree`, `BallTree`)
+* `metric`: distance metric (e.g. `Euclidean()`, `Chebyshev()`)
+* `searchstructure_kwargs...`: additional keyword arguments passed to `searchstructure`
+
+For developing a new `BasinOfAttraction` subtype extend the internal function `map_to_domain`,
+and ensure that the basins can be indexed by the returned value.
+
+"""
+function map_to_basin(BoA::BasinsOfAttraction, point; kwargs...)
+    n = map_to_domain(BoA, point; kwargs...)
+    return extract_basins(BoA)[n]
+end
+
+# Map point to index of the nearest member of the domain of the basin of attraction
+map_to_domain(BoA::ArrayBasinsOfAttraction, point) = basin_cell_index(point, BoA.grid) # Nearest grid cell
+function map_to_domain(BoA::SampledBasinsOfAttraction, point; tree = KDTree, metric = Euclidean(), ss_kwargs...) 
+    ss = searchstructure(tree, BoA.sampled_points, metric, ss_kwargs...) # Nearest sampled point
+    idx, _ = nn(ss, point, args...; kwargs...)
+    return idx
```

Comment:
This is uber inefficient :D because constructing the KD tree is the most expensive part of the computation. I think, we should construct the KDtree and store it as a field in the _construction_ of the Sampled basins type. Then, remove the option of tree and metric as arguments. They are options for the constructor of the sampled basins instead. We just always initialize a tree even if it is not required.

Later this can be optimized by having the tree field undef, and initializing it on demand if this function is called for the first time, but this can be done later.